### DISCUSSION TITLE
Graphql mutation validation errors

### DISCUSSION THREAD

**Discussion by gkielwasser** [directus/directus#8007] [Archive]
Hi,

I'm not sure what is the proper way of handling errors while trying to create a new user.

1- Data related: I would like to add a unique email constraint and to be able to read the appropriate error if it fails in the front application.
2- Input related: I would like to ensure strengthness of the password. I'm not sure but it looks like for this input validation I have to add a regexp in the field validation section in the role. Maybe I can then read an error key from the api response.

Currently, it look likes I only receive a boolean that tell me if the creation is successfull, otherwise I get null.

```
mutation signup($first_name: String!, $last_name: String!, $email: String!, $password: String!) {
  create_users_item(data: { first_name: $first_name, last_name: $last_name, email: $email, password: $password } )
}
```
Here is my error response. I have no clue reading the response from my front application why it fails.
```
{ "data": { "create_users_item": null } }
```

Thank you for feeback about those kind of validation

**Comment by rijkvanzanten** [comment]
I think this might've been resolved in https://github.com/directus/directus/issues/7651

**Reply by gkielwasser** [reply]
Ah thank you, yep this one partially solve! In the first case I have described I should have received a message 'Field "email" has to be unique'. It is not yet fixed in rc-92, it might be the future release.

Is it on purpose the original error code 'RECORD_NOT_UNIQUE' is not returned by the error formatter ?

From the front, there is no clean way to translate or at least react properly without knowing the original error code. I'm not sure how I could gave a user friendly feedback as 'This email is already used', 'This email format is invalid'.

For now we only see we those data
```
{
  "collection": "directus_users",
  "field": "email",
  "invalid": "already@taken.email"
}
```

Thanks!

**Reply by rijkvanzanten** [reply]
> It is not yet fixed in rc-92, it might be the future release.

That's correct. It's marked as fixed in release rc.93 👍🏻 

> Is it on purpose the original error code 'RECORD_NOT_UNIQUE' is not returned by the error formatter ?

No. GraphQL's `GraphQLFormattedError` type doesn't include a "code" flag, so GraphQL itself strips out the code in that stage. We could try injecting the original error code under _extensions_ 🤔 Wanna open a PR for that?

**Reply by gkielwasser** [reply]
Yep nice one !
Here is a [PR](https://github.com/directus/directus/pull/8034)
